client/memez/projects/memez/src/app/stores/entities/post.store.ts

```diff
@@ -1,69 +1,96 @@
+import {action, computed, observable} from 'mobx-angular';
+import {RootStore}                    from '../root.store';
+import {IPost}                        from '../../../../../../../../shared/types/Entities/IPost';
+import {autorun}                      from 'mobx';
 
-import * as dayjs from 'dayjs';
-import {autorun} from 'mobx';
-import {MOCK_POSTS} from '../../../../../../../../shared/mock/MOCK_POSTS';
-import {ILike} from '../../types/Entities/ILike';
 
 @Injectable({
   providedIn: 'root'
 })
 export class PostStore {
   @observable posts: IPost[] = [];
-  useMock: boolean = true;
+
+  /*useMock: boolean = false;*/
 
   constructor(
     public root: RootStore
   ) {
     this.root.ps = this;  //self-registration at root store
     window['postStore'] = this;
+    autorun(() => {
+
+    })
+
+
+  }
+
+  @computed get reversedPosts() {
+    return this.posts
+      .slice()
+      .reverse()
+  }
 
+  @computed get userPosts() {
+    const userPosts: IPost[] = this.posts
+      .filter(post => post.postedBy._id === this.root.log.currentUser._id)
+    return userPosts.reverse()
   }
 
+  
+
+
+
   @action
   async getPosts(): Promise<IPost[]> {
-    if (this.useMock) {
-      return this.posts = MOCK_POSTS;
-    } else {
-      this.posts = await this.root
-        .postAdapter
-        .getPosts();
-    }
+    /* if (this.useMock) {
+     return this.posts = MOCK_POSTS;
+     } */
+
+    this.posts = await this.root
+      .postAdapter
+      .getPosts();
+
+   /* this.posts
+      .map(async post => {
+        await this.root
+          .likeStore
+          .getPostLikes(post)
+      })*/
+
+    return this.posts;
+
   }
 
-  @action post(str: string) {
-    let newPost: IPost = {
-      id: 1,
-      user_id: 5,
-      content: str,
-      date: dayjs().format('DD.MM.YY'),
-      time: dayjs().format('HH:mm'),
-      likes: []
+  @action
+  async createPost(content: string) {
+    let postInput = {
+      user_id: this.root.log.currentUser._id,
+      content: content,
     }
-    this.posts.unshift(newPost);
-    this.root.log.currentUser.posts.push(newPost)
 
+    await this.root.postAdapter.createPost(postInput)
+
+    await this.getPosts()
 
-  }
 
-  @action deletePost(post: IPost) {
-    let index = this.posts.indexOf(post);
-    this.posts.splice(index, 1);
   }
 
-  @action postLiked(post: IPost, newLike: ILike) {
-    // Find if the current user has already liked the post
-    let found = post.likes.find(like => like.user_id === newLike.user_id);
-    if (found) {
-      // If found unlike
-      let index = post.likes.indexOf(found);
-      post.likes.splice(index, 1);
+  @action
+  async deletePost(post: IPost) {
+    if (post.postedBy._id === this.root.log.currentUser._id) {
+      await this.root.postAdapter.deletePost(post._id);
+
+      await this.getPosts()
     } else {
-      // if not found like
-      post.likes.push(newLike);
+      alert('This is not your post to DELETE ! ! ! ')
```

Comment:
already told you, no reason to show a button to delete if the post doesnt belong to the current user